MapboxNavigation/Style.swift

```diff
@@ -209,6 +210,95 @@ public class ResumeButton: UIControl {
     }
 }
 
+/// :nodoc:
+@objc(MBSpeedLimitSign)
+open class SpeedLimitSign: UIView {
+    
+    private static let textInsets = UIEdgeInsets(top: 6, left: 6, bottom: 6, right: 6)
+    
+    lazy var label: SpeedLimitLabel = .forAutoLayout()
+    
+    let defaultText = NSLocalizedString("SPEED_LIMIT", bundle: .mapboxNavigation, value: "Speed\nLimit\n", comment: "Speed limit sign main text.")
+    let paragraphStyle = NSMutableParagraphStyle()
+    
+    var speedLimit: SpeedLimit? {
+        didSet {
+            guard let speedLimit = speedLimit else { return }
+            let fontSize = UIFont.systemFontSize
+            
+            let strings = NSMutableAttributedString()
+            
+            strings.append(NSMutableAttributedString(string: defaultText.uppercased(), attributes: [
+                .paragraphStyle: paragraphStyle,
+                .font: UIFont.systemFont(ofSize: fontSize * 0.9)
+                ]))
+            
+            strings.append(NSMutableAttributedString(string: String(describing: speedLimit.speed), attributes: [
```

Comment:
Use a NumberFormatter so that, for example, a fractional speed limit uses the correct decimal separator.

Reply:
We should probably omit fractions huh?

Reply:
So if the speed limit is [7½&nbsp;mph](http://overpass-turbo.eu/s/xfT), would we round up or down?

[![7½](http://www.alastaircaldwell.com/Alastair's_Blog_files/03%20TransAm%202015/speed-sign.jpg)](http://www.alastaircaldwell.com/alastairs-blog/transam-rally-june-july-2015)

Reply:
More to the point, once we add [an Arabic localization](https://www.transifex.com/mapbox/mapbox-navigation-ios/language/ar/), we’d use Eastern Arabic numerals:

[![](https://upload.wikimedia.org/wikipedia/commons/4/4c/UAE_Speed_Limit_-_60_kmh.svg)](https://commons.wikimedia.org/wiki/File:UAE_Speed_Limit_-_60_kmh.svg)